smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/Utils.kt

```diff
@@ -19,7 +19,7 @@ fun String.removeSurroundingBackticks() = removeSurrounding("`", "`")
  */
 fun swiftEnumCaseName(name: String?, value: String, shouldBeEscaped: Boolean = true): String {
     val resolvedName = name ?: value
-    var enumCaseName = CaseUtils.toCamelCase(resolvedName.replace(Regex("[^a-zA-Z0-9_ ]"), ""))
+    var enumCaseName = CaseUtils.toCamelCase(resolvedName.replace(Regex("[^a-zA-Z0-9_ -]"), ""))
```

Comment:
The `CaseUtils::toCamelCase` also uses the dash (`-`) as one of the delimiters for camel case condition.